test/integration/api/user-preferences.js

```diff
@@ -0,0 +1,119 @@
+const { testService } = require('../setup');
+
+describe('api: user-preferences', () => {
+
+  it('should not allow storing preferences for nonexistent projects', testService(async (service) => {
+    const asAlice = await service.login('alice');
+
+    await asAlice.put('/v1/user-preferences/project/123/someProperty')
+      .send({ propertyValue: 'someValue' })
+      .expect(400); // project 123 doesn't exist
+  }));
+
+  it('PUT/DELETE preference storage operations', testService(async (service) => {
+    const asAlice = await service.login('alice');
+
+    await asAlice.put('/v1/user-preferences/site/someSimpleSitePref')
+      .send({ propertyValue: true })
+      .expect(201);
+
+    await asAlice.put('/v1/user-preferences/site/someComplexSitePref')
+      .send({ propertyValue: { 1: 2, 3: 4 } })
+      .expect(201);
+
+    // overwrite the previous
+    await asAlice.put('/v1/user-preferences/site/someComplexSitePref')
+      .send({ propertyValue: [1, 2, 3] })
+      .expect(201);
+
+    await asAlice.put('/v1/user-preferences/site/toBeDeletedPref')
+      .send({ propertyValue: 'troep' })
+      .expect(201);
+
+    await asAlice.put('/v1/user-preferences/project/1/someSimpleProjectPref')
+      .send({ propertyValue: false })
+      .expect(201);
+
+    await asAlice.put('/v1/user-preferences/project/1/someComplexProjectPref')
+      .send({ propertyValue: [1, 2, 'many'] })
+      .expect(201);
+
+    // make a new project (which should get ID 2 with current fixtures)
+    await asAlice.post('/v1/projects')
+      .send({ name: 'preferences test project' })
+      .expect(200)
+      .then(({ body }) => {
+        body.id.should.eql(2);
```

Comment:
when you make a new project in a test, because it may not be id 2. 

the test suite runs tests in 2 modes:
1. starts a transaction, runs the test, and then rolls back the transaction (to reset the database to the initial state)
2. starts the db from scratch, runs all the migrations, applies the fixtures, then carries on with the test (the migrations are slow, which is why this way isn't used for all the tests)

these modes, or maybe the interaction between the two of them, is how the new IDs for certain resources aren't always what you expect.